docs/sui-xcall.md

```diff
@@ -0,0 +1,90 @@
+
+### 4. Connection Registration
+
+#### Changes
+- xCall and centralized connection are now the same entity.
+- For multiple connections, admin can register the connection from xCall with a connection ID in the format `centralized-...` and a relayer. The connection address equivalent to other chain is the connection-id here.
+
+#### Connection Cap
+- A connection cap is generated and transferred to the relayer.
+- The relayer uses this cap to gain admin access to the centralized connection.
+- Admin rights are tied to the possession of the cap.
+
+#### Relayer Changes
+- To change the relayer, the connection cap can be transferred externally.
+
+## Upgrading Packages
+
+### Immutability and Versioning
+
+#### Sui Packages
+- Sui packages are immutable, meaning upgrading a package retains the original package, allowing both old and new packages to be used.
+- New features added in upgrade require the new package, while the old package can still operate using the same storage.
+
+#### Versioning
+- Introduced versioning in xcall and balanced to prevent the use of old packages after critical bug fixes or breaking changes.
+-  Upgrades involve updating the storage version and the constant version in the package. Each entry function enforces the version to ensure the old package is not used.
+
+### Event Handling
+
+#### Event Immutability
+- Events are immutable objects and won't upgrade to a newer package ID.
+- Events added in an upgrade must be tracked from the new package ID, while previous events are tracked from the old package ID.
+
+#### Event Listening
+- To listen to events added during an upgrade, all package IDs must be tracked.
+- Establishing websocket connections with each package ID is cumbersome, so adding events post-deployment is not recommended.
+
+### Dependency Upgrades:
+   - **Implementation**: If any dependencies are upgraded, the corresponding packages must also be upgraded to utilize the new features from these dependencies.
+   - **Process**: For instance, if the `rlp` package is upgraded, `xcall` should be upgraded to point to the new `rlp`, and then `balanced` should be upgraded to point to the new xCall to incorporate new features. This cascading upgrade ensures all improvements are integrated but increases the complexity of the development process.
+
+### Dapp Package Structure
+
+#### Balanced Package
+- **Modules**: Asset Manager, xCall Manager, Balanced Dollar.
+- **Identifiers**: Each package has one package ID and three cap IDs (one for each module). Cap IDs are akin to contract addresses in other chains.
+- **Usage**: Cap IDs are used for configuring Balanced in other chains, while the package ID is used for function calls from the Sui side.
+
+## Conclusion
+
+The updated implementation of the xcall deviates from the original xCall architecture but retains functional equivalence. 
+
```

Comment:
one thing that we missed here is usage of ticket ,its working principal and why its necessary. Secondly we the standards agreed upon like using dapp_cap_id in hex format for addresses.

Reply:
it would be nice to have some info on deployment process, maintainence of separate branches for deployment and its workflow.